tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs

```diff
@@ -2064,6 +2065,71 @@ public void MoStemMsa_EqualMsa()
 		}
 
+		/// <summary>
+		///
+		/// </summary>
+		[Test]
+		public void OnRefresh_SetNullInflFeatsOAWhenInvalidFeature()
+		{
+			ILangProject lp = Cache.LangProject;
+			ILexEntry le = MakeLexEntry("xyzTest1", "xyzDefn1.1");
+
+			IMoInflAffMsa infAffixMsa1 = Cache.ServiceLocator.GetInstance<IMoInflAffMsaFactory>().Create();
+			IMoInflAffMsa infAffixMsa2 = Cache.ServiceLocator.GetInstance<IMoInflAffMsaFactory>().Create();
+			IMoStemMsa stemMsa = Cache.ServiceLocator.GetInstance<IMoStemMsaFactory>().Create();
+			;
+			IMoDerivAffMsa derivAffixMsa = Cache.ServiceLocator.GetInstance<IMoDerivAffMsaFactory>().Create();
+			IMoUnclassifiedAffixMsa unclassifiedAffixMsa =
+				Cache.ServiceLocator.GetInstance<IMoUnclassifiedAffixMsaFactory>().Create();
+
+			le.MorphoSyntaxAnalysesOC.Add(infAffixMsa1);
+			le.MorphoSyntaxAnalysesOC.Add(infAffixMsa2);
+			le.MorphoSyntaxAnalysesOC.Add(stemMsa);
+			le.MorphoSyntaxAnalysesOC.Add(derivAffixMsa);
+			le.MorphoSyntaxAnalysesOC.Add(unclassifiedAffixMsa);
+
+			// Verify we fail comparing on different MSA types.
+			Assert.IsFalse(infAffixMsa1.EqualsMsa(stemMsa));
+			Assert.IsFalse(infAffixMsa1.EqualsMsa(derivAffixMsa));
+			Assert.IsFalse(infAffixMsa1.EqualsMsa(unclassifiedAffixMsa));
+
+			// Verify that infAffixMsa1 equals itself.
+			Assert.IsTrue(infAffixMsa1.EqualsMsa(infAffixMsa1), infAffixMsa1 + " - should equal itself.");
+
+			// Verify that infAffixMsa1 equals infAffixMsa2
+			Assert.IsTrue(infAffixMsa1.EqualsMsa(infAffixMsa2),
+				infAffixMsa1.ToString() + " - should equal - " + infAffixMsa2);
+
+			// compare with on different PartOfSpeech
+			ILcmOwningSequence<ICmPossibility> posSeq = lp.PartsOfSpeechOA.PossibilitiesOS;
+			IPartOfSpeech pos1 = Cache.ServiceLocator.GetInstance<IPartOfSpeechFactory>().Create();
+			posSeq.Add(pos1);
+			IPartOfSpeech pos2 = Cache.ServiceLocator.GetInstance<IPartOfSpeechFactory>().Create();
+			posSeq.Add(pos2);
+
+			infAffixMsa1.PartOfSpeechRA = pos1;
+			infAffixMsa2.PartOfSpeechRA = pos2;
+			Assert.IsTrue(infAffixMsa1.PartOfSpeechRA != infAffixMsa2.PartOfSpeechRA, "msa POSes should not be equal.");
+			Assert.IsFalse(infAffixMsa1.EqualsMsa(infAffixMsa2),
+				"infAffixMsa1 should not be equal to infAffixMsa2 due to different POS");
+
+			// compare different InflFeats
+			XmlDocument doc = new XmlDocument();
+			doc.LoadXml(MoreCellarTests.ksFS1);
+			XmlNode itemNeut = doc.SelectSingleNode("/item/item[3]");
+			XmlNode itemFem = doc.SelectSingleNode("/item/item[2]");
+			var featStruc1 = Cache.ServiceLocator.GetInstance<IFsFeatStrucFactory>().Create();
+			var featStruc2 = Cache.ServiceLocator.GetInstance<IFsFeatStrucFactory>().Create();
+			infAffixMsa1.InflFeatsOA = featStruc1;
+			infAffixMsa2.InflFeatsOA = featStruc2;
+			infAffixMsa1.InflFeatsOA.AddFeatureFromXml(itemNeut, Cache.LangProject.MsFeatureSystemOA);
+			infAffixMsa2.InflFeatsOA.AddFeatureFromXml(itemFem, Cache.LangProject.MsFeatureSystemOA);
+
+			// reset POS
```

Comment:
Change comment to // change to POS with incompatible features

Reply:
Ok